design/KEPS/extensions/proposal.md

```diff
@@ -0,0 +1,189 @@
+| Status_code      |                                                                                   |
+| Flags            | optional, defaults to “”                                                          |
+
+### Kiali_ext_response_time_seconds
+
+- histogram
+- Buckets are prom defaults: [ .005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10 ]
+
+| Attribute        | Notes                                                                             |
+| ---------------- | --------------------------------------------------------------------------------- |
+| Extension        | configured extension name                                                         |
+| Protocol         | http \| grpc                                                                      |
+| Source_cluster   |                                                                                   |
+| Source_namespace |                                                                                   |
+| Source_name      |                                                                                   |
+| Source_is_root   | this source is rooted in the kiali graph: true \| false                           |
+| Reporter         | source \| dest \| combined - “combined” is preferred (i.e. only 1 TS per request) |
+| Reporter_id      | pod id or other unique discriminator                                              |
+| Dest_cluster     |                                                                                   |
+| Dest_namespace   |                                                                                   |
+| Dest_name        |                                                                                   |
+| Security         | none \| mTLS                                                                      |
+| Status_code      |                                                                                   |
+| Flags            | optional, defaults to “”                                                          |
+
+## Kiali Extension Configuration
+
+The primary extension code, a graph appender, is generic and owned by Kiali. The Extensions will look for and process extension data for each enabled extension registered in the Kiali CR. The appender would likely be always-on, and just do nothing if no extensions are configured or enabled.
+
+```
+spec:
+  extensions:
+    - enabled: true | false (default)
+      name: <extension name>
+```
+
+## Kiali Extension UI URL linking
+
+While the intent of Kiali extensions is to be able to extend the Kiali graph beyond the mesh, the information about those extended services will be limited. Users may want to investigate further. If An Extension provide has its own console UI, it can ensure that the Kiali user can link to that UI. For extension nodes, the graph side panel will provide a link to the external UI, if discovered. There are two mechanisms:
```

Comment:
```suggestion
While the intent of Kiali extensions is to be able to extend the Kiali graph beyond the mesh, the information about those extended services will be limited. Users may want to investigate further. If an Extension provider has its own console UI, it can ensure that the Kiali user can link to that UI. For extension nodes, the graph side panel will provide a link to the external UI, if discovered. There are two mechanisms:
```